app/assets/javascripts/d3/mekko.coffee

```diff
@@ -247,17 +250,23 @@ D3.mekko =
     display_legend: =>
       $(@container_selector()).find('.legend').remove()
 
-      legend_columns = if @carrier_list.length > 9 then 3 else 2
-
-      series = _.filter(@carrier_list.models, (s) ->
-        Math.abs(s.total_value()) > 1e-7
-      )
-
       @draw_legend
-        series: series
+        series: @prepare_legend_items()
         width: @width
         vertical_offset: @series_height + @label_height
-        columns: legend_columns
+        columns: 2
+
+    prepare_legend_items: =>
```

Comment:
Since Mekko charts only show future values, I don't think it's necessary to check the present value. I'm not sure why the legend list was changed from `_.filter(@carrier_list.models, (s) -> Math.abs(s.total_value()) > 1e-7)`; it's causing this because it enumerates through each series instead of each carrier:

<img width="595" alt="screen shot 2018-04-11 at 09 12 43" src="https://user-images.githubusercontent.com/4383/38605490-5b0e045a-3d6b-11e8-8b30-b747066ef819.png">

Reply:
The ideas was to reverse the output of the legend items. Funny that I didn't saw the multiple legend items. I'll fix it.